test/unit/tools/test_tool_deps.py

```diff
@@ -719,25 +719,69 @@ def _first_conda_resolver_options(dm):
 
 
 @contextmanager
-def __environ(values, remove=[]):
+def __environ(values, keys_to_remove=[]):
```

Comment:
I distribute bonus points for eliminating mutable default args, see https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments. The way you use it here it's fine,
but better to just say `keys_to_remove=None` and then 
```
if keys_to_remove is None:
    keys_to_remove = []
```
in the function body.

Reply:
@mvdbeek Yes! (and there's no need to create [] in the body: if it's none, we never use it - so a simple check will be sufficient). Thx!

Reply:
Done.